### PR TITLE
feat: project id move

### PR DESCRIPTION
**Pull request by mhagel**
## Description

Closes: regen-network/regen-registry#1442

- Moves Project Id out of the "Additional Info" section, and puts it at the top of the page for on-chain projects.
- If there is no `metadata[schema:name]`, uses the projectId for the header

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. https://deploy-preview-1678--regen-registry.netlify.app/projects/C01-001
2. https://deploy-preview-1678--regen-registry.netlify.app/projects/kasigau
3. https://deploy-preview-1678--regen-registry.netlify.app/projects/C45-005

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/components/molecules/ProjectPageMetadata/ProjectPageMetadata.tsx [hunk 1]
```diff
@@ -67,6 +62,23 @@ const ProjectPageMetadata: React.FC<React.PropsWithChildren<MetadataProps>> = ({
...
+              <MetaDetail
+                label="project activity"
+                data={
+                  <ArrowLink
+                    href={projectActivity?.['schema:url']?.['@value'] || ''}
+                    label={projectActivity?.['schema:name']}
+                  />
+                }
+              />
+            )}
```
Comment (mhagel) [inline web-registry/src/components/molecules/ProjectPageMetadata/ProjectPageMetadata.tsx:81]: Just moved these up to look like mocks

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx [hunk 2]
```diff
@@ -149,8 +155,13 @@ function ProjectTopSection({
                 mt: 2.5,
               }}
             >
-              {!metadata && loading && <Skeleton variant="text" height={124} />}
-              {!onChainProject && (
+              {onChainProjectId ? (
+                projectName && (
```
Comment (mhagel) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:159]: if this is on-chain, show projectId label IFF projectName is displayed above. 

If projectName is not displayed above, the header falls back to `Project ${onChainProjectId}`. We don't want to display projectId here again in this case.

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx [hunk 3]
```diff
@@ -203,7 +214,7 @@ function ProjectTopSection({
               sx={{ mt: [2, 4], py: [2, 6] }}
             />
           </Link>
-          <ProjectPageMetadata metadata={metadata} projectId={projectId} />
+          {onChainProjectId && <ProjectPageMetadata metadata={metadata} />}
```
Comment (mhagel) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:217]: Don't show metadata section on legacy projects.

File: web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx [hunk 4]
```diff
@@ -86,7 +86,7 @@ function ProjectDetails(): JSX.Element {
 
   const onChainProjectId = isOnChainId
     ? projectId
-    : dataByHandle?.projectByHandle?.onChainId;
+    : dataByHandle?.projectByHandle?.onChainId ?? undefined;
```
Comment (mhagel) [inline web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx:89]: TS compiler found null as a possibility here, so explicitly falling back to `undefined`.

**Comment by mhagel** [comment]
@erikalogie ready for feedback

**Comment by erikalogie** [comment]
Looks good!

**Review by flagrede — APPROVED** [review]
Looks good !

**Review by blushi — APPROVED** [review]

**Comment by mhagel** [comment]
@mergifyio backport hambach redwood

**Pull request merged**